janus.c

```diff
@@ -435,28 +435,28 @@ janus_session *janus_session_create(guint64 session_id) {
 	session->last_activity = janus_get_monotonic_time();
 	janus_mutex_init(&session->mutex);
 	janus_mutex_lock(&sessions_mutex);
-	g_hash_table_insert(sessions, GUINT_TO_POINTER(session_id), session);
+	g_hash_table_insert(sessions, janus_uint64_dup(session->session_id), session);
 	janus_mutex_unlock(&sessions_mutex);
 	return session;
 }
 
 janus_session *janus_session_find(guint64 session_id) {
 	janus_mutex_lock(&sessions_mutex);
-	janus_session *session = g_hash_table_lookup(sessions, GUINT_TO_POINTER(session_id));
+	janus_session *session = g_hash_table_lookup(sessions, &session_id);
```

Comment:
It is expected, as we're using `g_hash_table_new_full(g_int64_hash, g_int64_equal`, which uses a dedicated method is used to compare them to find the right one. The reason why we're allocating memory for the key is because the pointer you use for inseerting is the one that is used for actually addressing the key: all other calls on the key instead (lookup, remove) are hashed and compared using the specified methods, and so can be temporary. If you use a temporary `guint64` and use a pointer to that for inserting, as soon as the calling method returns the key points to garbage and the value is lost.